app-devel/intel-oneapi-basekit/autobuild/postinst

```diff
@@ -0,0 +1,2 @@
+echo "To setup the environment properly and use Intel oneAPI toolkit, execute:"
+echo ". /usr/lib/intel/oneapi/setvars.sh"
```

Comment:
This is probably poor design for user interface - can we do this in the profile script?

Reply:
Technically, we can set it up in the profile script so the user doesn't need to run it manually.
However, there might be two problems.

1. Intel's OpenCL headers may conflict with NVIDIA's OpenCL headers. The Intel DPCPP compiler's building guide specifically mentions this.
```
...
Non-standard CUDA location:

If the CUDA toolkit is installed in a non-default location on your system, two considerations must be made.

Firstly, do not add the toolkit to your standard environment variables (PATH, LD_LIBRARY_PATH), as to do so will create conflicts with OpenCL headers.
...
```

2. oneAPI toolkit is based on LLVM and has some runtime libraries with the same name as the upstream LLVM, such as ```libomptarget.so```. After setting up the environment, the script will add oneAPI's directories to ```LD_LIBRARY_PATH``` and cause conflicts. The programs compiled by LLVM or supposed to use LLVM's runtime will load Intel's runtime instead. This mismatching may lead to unexpected results.

So, I guess it may be better to let the individual script run on demand. Arch Linux also handles the package in the same way.

Reply:
> Technically, we can set it up in the profile script so the user doesn't need to run it manually. However, there might be two problems.
> 
>     1. Intel's OpenCL headers may conflict with NVIDIA's OpenCL headers. The Intel DPCPP compiler's building guide specifically mentions this.
> 
> 
> ```
> ...
> Non-standard CUDA location:
> 
> If the CUDA toolkit is installed in a non-default location on your system, two considerations must be made.
> 
> Firstly, do not add the toolkit to your standard environment variables (PATH, LD_LIBRARY_PATH), as to do so will create conflicts with OpenCL headers.
> ...
> ```
> 
>     2. oneAPI toolkit is based on LLVM and has some runtime libraries with the same name as the upstream LLVM, such as `libomptarget.so`. After setting up the environment, the script will add oneAPI's directories to `LD_LIBRARY_PATH` and cause conflicts. The programs compiled by LLVM or supposed to use LLVM's runtime will load Intel's runtime instead. This mismatching may lead to unexpected results.
> 
> 
> So, I guess it may be better to let the individual script run on demand. Arch Linux also handles the package in the same way.

I see, I find two ways of doing this:

1. Keep with your current logic of prompting the user at postinst. I would, however, reference app-admin/gru

> Technically, we can set it up in the profile script so the user doesn't need to run it manually. However, there might be two problems.
> 
>     1. Intel's OpenCL headers may conflict with NVIDIA's OpenCL headers. The Intel DPCPP compiler's building guide specifically mentions this.
> 
> 
> ```
> ...
> Non-standard CUDA location:
> 
> If the CUDA toolkit is installed in a non-default location on your system, two considerations must be made.
> 
> Firstly, do not add the toolkit to your standard environment variables (PATH, LD_LIBRARY_PATH), as to do so will create conflicts with OpenCL headers.
> ...
> ```
> 
>     2. oneAPI toolkit is based on LLVM and has some runtime libraries with the same name as the upstream LLVM, such as `libomptarget.so`. After setting up the environment, the script will add oneAPI's directories to `LD_LIBRARY_PATH` and cause conflicts. The programs compiled by LLVM or supposed to use LLVM's runtime will load Intel's runtime instead. This mismatching may lead to unexpected results.
> 
> 
> So, I guess it may be better to let the individual script run on demand. Arch Linux also handles the package in the same way.

Got it. I see two ways of handling this:

1. Keep with the "prompt at postinst" logic - but I would strongly recommend that you reference `app-productivity/taskwarrior` and use [debconf](http://www.fifi.org/doc/debconf-doc/tutorial.html) to draw a prompt.
2. Make this script more readily accessible (i.e., installed in `/usr/bin`).
3. The most complicated but perhaps ideal way - make a switcher tool for different LLVM development runtimes.

Reply:
1. The message is added via `debconf` as proposed.
2. A symbol link `setup_intel_oneapi.sh` pointing `/usr/lib/intel/oneapi/setvars.sh` is added to `/usr/bin`. User can run `. setup_intel_oneapi.sh` anywhere to set up oneAPI environment.